src/Runtime/OMSort.inc

```diff
@@ -0,0 +1,453 @@
+    break;
+  case ONNX_TYPE_INT32:
+    compFunc = ascending ? compareFunction(Int32, int32, Ascending)
+                         : compareFunction(Int32, int32, Descending);
+    break;
+  case ONNX_TYPE_UINT64:
+    compFunc = ascending ? compareFunction(Uint64, uint64, Ascending)
+                         : compareFunction(Uint64, uint64, Descending);
+    break;
+  case ONNX_TYPE_INT64:
+    compFunc = ascending ? compareFunction(Int64, int64, Ascending)
+                         : compareFunction(Int64, int64, Descending);
+    break;
+  case ONNX_TYPE_FLOAT:
+    compFunc = ascending ? compareFunction(Float, float, Ascending)
+                         : compareFunction(Float, float, Descending);
+    break;
+  case ONNX_TYPE_DOUBLE:
+    compFunc = ascending ? compareFunction(Double, double, Ascending)
+                         : compareFunction(Double, double, Descending);
+    break;
+  default:
+    assert(false && "unexpected data type in getCompareFunction");
+  }
+  return compFunc;
+}
+
+uint64_t getDataSize(OM_DATA_TYPE dataType) {
+  uint64_t size = 0;
+  switch (dataType) {
+  case ONNX_TYPE_BOOL:
+    size = sizeof(bool);
+    break;
+  case ONNX_TYPE_UINT8:
+    size = sizeof(uint8_t);
+    break;
+  case ONNX_TYPE_INT8:
+    size = sizeof(int8_t);
+    break;
+  case ONNX_TYPE_UINT16:
+    size = sizeof(uint16_t);
+    break;
+  case ONNX_TYPE_INT16:
+    size = sizeof(int16_t);
+    break;
+  case ONNX_TYPE_UINT32:
+    size = sizeof(uint32_t);
+    break;
+  case ONNX_TYPE_INT32:
+    size = sizeof(int32_t);
+    break;
+  case ONNX_TYPE_UINT64:
+    size = sizeof(uint64_t);
+    break;
+  case ONNX_TYPE_INT64:
+    size = sizeof(int64_t);
+    break;
+  case ONNX_TYPE_FLOAT:
+    size = sizeof(float);
+    break;
+  case ONNX_TYPE_DOUBLE:
+    size = sizeof(double);
+    break;
+  default:
+    assert(false && "unexpected data type in getDataSize");
+  }
+  return size;
+}
+
+void omTensorSort(OMTensor *orderTensor, const OMTensor *inputTensor,
+    uint64_t axis, uint64_t ascending, uint64_t algorithm) {
+  const OM_DATA_TYPE dataType = omTensorGetDataType(inputTensor);
+  const uint64_t rank = omTensorGetRank(inputTensor);
+  assert(rank <= 6 && "omTensorSort assumes rank <= 6");
+  assert(axis == (rank - 1) && "omTensorSort assumes axis == (rank - 1)");
+  const int64_t *inputShape = omTensorGetShape(inputTensor);
+  const int64_t *inputStrides = omTensorGetStrides(inputTensor);
+  assert(inputStrides[axis] == 1 && "omTensorSort assumes strides[axis] == 1");
+  void *orderPtr = omTensorGetDataPtr(orderTensor);
+  uint64_t *order = (uint64_t *)orderPtr;
+  void *dataPtr = omTensorGetDataPtr(inputTensor);
+  int64_t sort_elems = inputShape[axis];
+  // Sorting not necessary for empty array
+  if (sort_elems == 0)
+    return;
+
+  // Choose the appropriate compare function
+  compareFunctionType *compareElements =
+      getCompareFunction(ascending, dataType);
+  uint64_t datasize = getDataSize(dataType);
```

Comment:
We already have `OM_DATA_TYPE_SIZE[dataType]` for this in `OnnxDataType.h`.

Reply:
Fixed to use OM_DATA_TYPE[dataType] instead of the custom getDataSize function. Thank you!